Classes/Service/AttachmentService.php

```diff
@@ -25,30 +30,37 @@ public function initAttachments(array $attachments)
         /* @var \Mittwald\Typo3Forum\Domain\Model\Forum\Attachment */
         $objAttachments = new ObjectStorage();
 
-        foreach ($attachments as $attachmentID => $attachment) {
-            if ($attachment['name'] == '') {
-                continue;
-            }
-            $tmp_name = $_FILES['tx_typo3forum_pi1']['tmp_name']['attachments'][$attachmentID];
-            $mime_type = mime_content_type($tmp_name);
-
-            //Save in ObjectStorage and in file system
-            $attachmentObj = $this->objectManager->get(\Mittwald\Typo3Forum\Domain\Model\Forum\Attachment::class);
-            $attachmentObj->setFilename($attachment['name']);
-            $attachmentObj->setRealFilename(sha1($attachment['name'] . time()));
-            $attachmentObj->setMimeType($mime_type);
-
-            //Create dir if not exists
-            $tca = $attachmentObj->getTCAConfig();
-            $path = $tca['columns']['real_filename']['config']['uploadfolder'];
-            if (!file_exists($path)) {
-                mkdir($path, 0777, true);
-            }
+        foreach ($attachments as $attachment) {
+            foreach ($attachment as $attachmentID => $eachAttachment) {
```

Comment:
looks strange this double-loop, is it required?

Reply:
under investigation.